packages/common/src/chains/index.ts

```diff
@@ -1,16 +1,42 @@
-import { chainsType } from './../types'
+import { Chain, chainsType } from './../types'
+import mainnet from './mainnet.json'
+import ropsten from './ropsten.json'
+import rinkeby from './rinkeby.json'
+import kovan from './kovan.json'
+import goerli from './goerli.json'
 
-export const chains: chainsType = {
-  names: {
+/**
+ * @hidden
+ */
+export function _getInitializedChains(customChains?: Chain[]) {
+  const names: any = {
     '1': 'mainnet',
     '3': 'ropsten',
     '4': 'rinkeby',
     '42': 'kovan',
     '5': 'goerli',
-  },
-  mainnet: require('./mainnet.json'),
-  ropsten: require('./ropsten.json'),
-  rinkeby: require('./rinkeby.json'),
-  kovan: require('./kovan.json'),
-  goerli: require('./goerli.json'),
+  }
+  const chains: any = {
+    mainnet,
+    ropsten,
+    rinkeby,
+    kovan,
+    goerli,
+  }
+  if (customChains) {
+    for (const chain of customChains) {
+      const name = chain.name
+      names[chain.chainId.toString()] = name
+      chains[name] = chain
+    }
+  }
+
+  chains['names'] = names
```

Comment:
Hmm. So custom chains are treated on a different foot than "normal" chains. Can we either move these custom chains into the special field of `chains.names` or disallow the custom chain name `names`? (Very unlikely to happen, but it will probably mess up the internal logic if someone uses this name).

Reply:
Are you maybe misreading the code here (or I get your message wrong)? Custom chains are just normally added to the `names` dict here and the other entries are not affected in any way.

Generally I would want to move away from this `chains/index.ts` file altogether - therefore the deprecation notes. I would already want to move this to logic to the main `index.ts` file but then I kept it like this also for some backwards compatibility considerations.

Reply:
At line 30 it sets `chains[name]` to `chain` - so if you'd use the `name` "names", then it'd first set `chains.names` to the chain and then later overwrite it. Not sure if that causes problems?